fuel-tx/src/transaction/types/output.rs

```diff
@@ -181,7 +181,11 @@ impl Output {
     }
 
     pub fn message_nonce(txid: &Bytes32, idx: Word) -> Nonce {
-        (*Hasher::default().chain(txid).chain([idx as u8]).finalize()).into()
+        (*Hasher::default()
+            .chain(txid)
+            .chain(idx.to_bytes())
```

Comment:
If we are okay with supporting more than `255` receipts, then we also need to update the [specification](https://github.com/FuelLabs/fuel-specs/blob/master/src/identifiers/utxo-id.md#message-nonce)

Reply:
I think this was settled yesterday and we do want to support U16::MAX receipts. Would a PR changing the linked paragraph be enough? I recall there was some discussion about serialization as well.

Reply:
We need to update the paragraph and mention that we need to use canonical serialization for integers. 

We have a separate issue to add a `u16::max` limit https://github.com/FuelLabs/fuel-vm/issues/620